tools/distpartitioning/parmetis_wrapper.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+import logging
+import os
+import platform
+import sys
+from pathlib import Path
+
+import constants
+from utils import read_json
+
+
+def run_parmetis_wrapper(params):
+    """Function to execute all the steps needed to run ParMETIS
+
+    Parameters:
+    -----------
+    params : argparser object
+        an instance of argparser class to capture command-line arguments
+    """
+    assert os.path.isfile(params.schema_file)
+    assert os.path.isfile(params.hostfile)
+
+    schema = read_json(params.schema_file)
+    graph_name = schema[constants.STR_GRAPH_NAME]
+    num_partitions = len(schema[constants.STR_NUM_NODES_PER_CHUNK][0])
+
+    # Trigger pre-processing step to generate input files for ParMETIS.
+    preproc_cmd = (
+        f"mpirun -np {num_partitions} -hostfile {params.hostfile} "
+        f"python3 tools/distpartitioning/parmetis_preprocess.py "
+        f"--schema_file {params.schema_file} "
+        f"--output_dir {params.preproc_output_dir}"
+    )
+    logging.info(f"Executing Preprocessing Step: {preproc_cmd}")
+    os.system(preproc_cmd)
+    logging.info(f"Done Preprocessing Step")
+
+    # Trigger ParMETIS for creating metis partitions for the input graph.
+    parmetis_install_path = "pm_dglpart3"
+    if params.parmetis_install_path is not None:
+        parmetis_install_path = os.path.join(
+            params.parmetis_install_path, parmetis_install_path
+        )
+    parmetis_nfiles = os.path.join(
+        params.preproc_output_dir, "parmetis_nfiles.txt"
+    )
+    parmetis_efiles = os.path.join(
+        params.preproc_output_dir, "parmetis_efiles.txt"
+    )
+    parmetis_cmd = (
+        f"mpirun -np {num_partitions} -hostfile {params.hostfile} "
+        f"{parmetis_install_path} {graph_name} {num_partitions} "
+        f"{parmetis_nfiles} {parmetis_efiles}"
+    )
+    logging.info(f"Executing ParMETIS: {parmetis_cmd}")
+    os.system(parmetis_cmd)
+    logging.info(f"Done ParMETIS execution step")
+
+    # Trigger post-processing step to convert parmetis output to the form
+    # acceptable by dist. graph partitioning pipeline.
+    parmetis_output_file = os.path.join(
+        os.getcwd(), f"{graph_name}_part.{num_partitions}"
+    )
+    postproc_cmd = (
+        f"python3 tools/distpartitioning/parmetis_postprocess.py "
+        f"--schema_file {params.schema_file} "
+        f"--parmetis_output_file {parmetis_output_file} "
+        f"--partitions_dir {params.partitions_dir}"
+    )
+    logging.info(f"Executing PostProcessing: {postproc_cmd}")
+    os.system(postproc_cmd)
+    logging.info("Done Executing ParMETIS...")
+
+
+if __name__ == "__main__":
```

Comment:
how about checking whether `mpirun` and related dependencies are available at the very beginning?

Reply:
Done. We now check if the "mpirun" is found in the system's exec_path. If not found an assertion will fail and we will not go through the execution of this file.